apps/analytics/activity_reports.py

```diff
@@ -0,0 +1,117 @@
+        'operation_date': {'type': 'datetime'},
+        'subject': metadata_schema['subject'],
+        'keywords': metadata_schema['keywords'],
+        'group_by': {'type': 'list'},
+        'report': {'type': 'dict'},
+        'timestamp': {'type': 'datetime'},
+        'force_regenerate': {'type': 'boolean', 'default': False}
+    }
+    item_methods = ['GET', 'DELETE']
+    resource_methods = ['POST']
+
+    privileges = {'POST': 'activity_reports', 'DELETE': 'activity_reports', 'GET': 'activity_reports'}
+
+
+class ActivityReportService(BaseService):
+    # check if some fields are filled out before generating the report and initiate the filter
+    def check_for_fields(self, report):
+        terms = [
+            {"term": {"operation": report['operation']}}
+        ]
+        if report.get('subject'):
+            subjects = [subject['qcode'] for subject in report['subject']]
+            terms.append({'term': {'subject.qcode': subjects}})
+        if report.get('keywords'):
+            key = [x.lower() for x in report['keywords']]
+            terms.append({'term': {'keywords': key}})
+        if report.get('operation_date'):
+            op_date = items.ItemsService._format_date(report['operation_date'])
+            terms.append({'range': {'versioncreated': {'gte': op_date, 'lte': op_date}}})
+
+        return terms
+
+    # get the list of all items in the 4 repos
+    def get_items_list(self, query):
+            request = ParsedRequest
+            request.args = {'source': json.dumps(query), 'repo': 'archive,published,archived,ingest'}
+            items_list = list(get_resource_service('search').get(req=request, lookup=None))
+            return items_list
+
+    def search_items_without_groupping(self, report):
+        terms = []
+        terms.append({"term": {"task.desk": str(report['desk'])}})
+        terms.append(self.check_for_fields(report))
+
+        query = {
+            "query": {
+                "filtered": {
+                    "filter": {
+                        "bool": {"must": terms}
+                    }
+                }
+            }
+        }
+
+        items_list = self.get_items_list(query)
+        return {'items': len(items_list)}
+
+    def search_items_when_groupping(self, report):
+        result_list = []
+        desks = get_resource_service('desks').get(req=None, lookup={})
+
+        # return the filtered items for each desk
+        for desk in desks:
+            terms = []
+            terms.append({"term": {"task.desk": str(desk['_id'])}})
+            terms.append(self.check_for_fields(report))
+            query = {
+                "query": {
+                    "filtered": {
+                        "filter": {
+                            "bool": {"must": terms}
+                        }
+                    }
+                }
+            }
+            items_list = self.get_items_list(query)
+            item = {'desk': desk['name'], 'items': len(items_list)}
```

Comment:
Can we use `aggregates` on `task.desk` as count of items is required and it can be done in one query?